lib/openxml/has_properties.rb

```diff
@@ -0,0 +1,85 @@
+module OpenXml
+  module HasProperties
+
+    def self.included(base)
+      base.extend ClassMethods
+    end
+
+    module ClassMethods
+      def properties_tag(*args)
+        @properties_tag = args.first if args.any?
+        @properties_tag
+      end
+
+      def value_property(name, as: nil)
+        attr_reader name
+
+        properties[name] = (as || name).to_s
+
+        class_eval <<-CODE, __FILE__, __LINE__ + 1
+        def #{name}=(value)
+          property_key = "#{name}".to_sym
+          class_name = properties[property_key].split("_").map(&:capitalize).join
+          prop_module = Object.const_get class.name.split("::")[0...-2].push("Properties").join("::")
```

Comment:
I wonder why Rails has a more doesn't just use `Object.const_get`

https://github.com/rails/rails/blob/v5.1.3/activesupport/lib/active_support/inflector/methods.rb#L258-L287

What are they working around? ....

Reply:
My guess was that they were allowing for things other than fully-qualified names, e.g., `::ClassName` and the like...